### ISSUE TITLE
WebUI hammers my daemon

### ISSUE THREAD

**Issue by hsanjuan** [ipfs-shipyard/ipfs-webui#882]
When I open the webUI (maybe after clicking `Files`) it starts triggering hundreds `http://127.0.0.1:5001/api/v0/object/get` requests to the IPFS daemon and everything stops working well (my browser, my daemon, my computer). I feel things shouldn't be like this?

**Comment by olizilla** [comment]
Yes. We feel the same feels. There is a runaway cpu load issue caused by using the ipfs-geoip db ipfs/go-ipfs#5613 

We've dialled it down, but it could do less. In the short term I will have move the geolocation logic to just the peers page. It will still hammer the daemon, but only when you go looking at your peers list / map. There is #848 to track that.

Longer term we need to calm ipfs down. It should be possible to look up all the ipfs-geoip blocks for 800 ipv4 addresses, without causing the fans to spin.

**Comment by olizilla** [comment]
@hsanjuan web ui no longer tries to do the geolocation unless you are looking at the peers tab. Please take another look; https://webui.ipfs.io has the latest, or when go-ipfs v0.4.19 is released.

**Comment by hsanjuan** [comment]
How many `object/get` is this going to trigger at once (even if only in the peers tab)? I feel it's still not optimal that my daemon explodes when visiting a tab. Automatic parallel DHT lookups for b-tree objects for 500 peers might not be the best approach here?

I really like the idea of making B-trees storing the geo-ip information on ipfs though. It's very nice.

**Comment by olizilla** [comment]
I've dialled it right down, so we queue the requests and excute them one by one rather than in parallel. I plan to redesign the peers page this quarter so that it focuses on highlighting peers near you, and lets you connect to peers manually, rather than just "geolocate them all", to reduce the issue further.
